includes/datagrid-ref-confirm-action-and-validate-data.md

```diff
@@ -0,0 +1,187 @@
+---
+
+##### jQuery
+
+    <!--JavaScript-->
+    $(function(){
+        $("#{widgetName}Container").dx{WidgetName}({
+            // ...
+            ${{apiMember}}: function(e) {
+                const deferred = $.Deferred();
+                const promptPromise = DevExpress.ui.dialog.confirm("Are you sure?", "Confirm changes");
+                promptPromise.done((dialogResult) => {
+                    if (dialogResult) {
+                        let params = "";
+                        for (let key in e.newData) {
+                            params += `${key}=${e.newData[key]}&`;
+                        }
+                        params = params.slice(0, -1);
```

Comment:
to delete last '&'?

Reply:
Why not approach from Custom Data Source demo?

Reply:
Yes.

Reply:
You can pass e.newData as data to the $.ajax - it should convert object to string. 
https://api.jquery.com/jquery.ajax/